edn/edn.g4

```diff
@@ -0,0 +1,124 @@
+
+ 1. Redistributions of source code must retain the above copyright notice, this list of conditions
+ and the following disclaimer. 2. Redistributions in binary form must reproduce the above copyright
+ notice, this list of conditions and the following disclaimer in the documentation and/or other
+ materials provided with the distribution. 3. Neither the name of Tom Everett nor the names of its
+ contributors may be used to endorse or promote products derived from this software without specific
+ prior written permission.
+
+ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR
+ IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+ FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+ CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER
+ IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT
+ OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+grammar edn;
+program: value* EOF;
+value:
+	NilLiteral
+	| BooleanLiteral
+	| StringLiteral
+	| IntegerLiteral
+	| FloatingPointLiteral
+	| CharacterLiteral
+	| keyword
+	| Symbol
+	| tag
+	| list
+	| vector
+	| map
+	| set;
+
+//maybe add more tags here? did not get a lot of info
+tag: Hash Symbol;
+keyword:
+	Colon (
+		Symbol
+		| IntegerLiteral
+		| FloatingPointLiteral
+		| NilLiteral
+		| BooleanLiteral
+		// maybe a character literal here?
+	);
+list: LeftParenthesis value* RightParenthesis;
+vector: LeftBracket value* RightBracket;
+map: LeftBrace (value value)* RightBrace;
+set: HashedLeftBrace value* RightBrace;
+
+NilLiteral: 'nil';
+
+BooleanLiteral: 'true' | 'false';
+
+StringLiteral:
+	'"' (
+		~["\\]
+		| (
+			'\\' (
+				[ctrn\\"]
+				| [bf] //
+				| UnicodeCharacter
+			)
+		)
+	)* '"';
+
+IntegerLiteral: Int 'N'?;
+
+FloatingPointLiteral: Int ('M' | FractionalPart? Exponent?);
+
+CharacterLiteral:
+	'\\' (
+		'newline'
+		| 'return'
+		| 'space'
+		| 'tab'
+		| 'backspace'
+		| 'formfeed'
+		| UnicodeCharacter
+		| ~[ \r\n\t]
+	);
+
+LeftParenthesis: '(';
+RightParenthesis: ')';
+LeftBracket: '[';
+RightBracket: ']';
+LeftBrace: '{';
+RightBrace: '}';
+HashedLeftBrace: '#{';
+Hash: '#';
+Colon: ':';
+
+FractionalPart: '.' Digit+;
+fragment Exponent: E Digit+;
+fragment E: [eE] [+-]?;
+Int: [+-]? ('0' | [1-9] Digit*);
+fragment Digit: [0-9];
+fragment HexDigit: [a-fA-F0-9];
+fragment UnicodeCharacter:
+	'u' HexDigit HexDigit HexDigit HexDigit;
+
+Symbol: '/' | Name ('/' Name)?;
+
+Name: (( [-+.] (Alpha | Extra | Special)) | (Alpha | Extra)) (
+		Alpha
+		| Numeric
+		| Extra
+		| Special
+	)*;
+//TODO ADD more UTF-8 Compilant Characters
```

Comment:
Try unicode category, e.g. `\p{L}` for any letter

Reply:
Ah, forgot about that feature. Thank you! Will make a commit.